File: Codigo de los 4 lenguajes/js/EjemploClases.js
// EjemploClases.js
export default class EjemploClases {
  mostrar() {
    console.log(" Ejemplo: Clases reales ");

    class Persona {
      constructor(nombre, edad, activo) {
        this.nombre = nombre;
        this.edad = edad;
        this.activo = activo;
      }
      mostrarInfo() {
        console.log(`Persona  nombre: ${this.nombre}, edad: ${this.edad}, activo: ${this.activo}`);
      }
    }

    class Carro {
      constructor(marca, modelo, precio) {
        this.marca = marca;
        this.modelo = modelo;
        this.precio = precio;
      }
      mostrarInfo() {
        console.log(`Carro  marca: ${this.marca}, modelo: ${this.modelo}, precio: ${this.precio}`);
      }
    }

    class Animal {
      constructor(especie, nombre, domestico) {
        this.especie = especie;
        this.nombre = nombre;
        this.domestico = domestico;
      }
      mostrarInfo() {
        console.log(`Animal  especie: ${this.especie}, nombre: ${this.nombre}, domestico: ${this.domestico}`);
      }
    }

    class Libro {
      constructor(titulo, autor, paginas) {
        this.titulo = titulo;
        this.autor = autor;
        this.paginas = paginas;
      }
      mostrarInfo() {
        console.log(`Libro  titulo: ${this.titulo}, autor: ${this.autor}, paginas: ${this.paginas}`);
      }
    }

    class Computador {
      constructor(marca, ramGb, encendido) {
        this.marca = marca;
        this.ramGb = ramGb;
        this.encendido = encendido;
      }
      mostrarInfo() {
        console.log(`Computador  marca: ${this.marca}, RAM: ${this.ramGb}GB, encendido: ${this.encendido}`);
      }
    }

    const p = new Persona("Camilo", 20, true);
    const c = new Carro("Toyota", 2020, 35000.0);
    const a = new Animal("Perro", "Firulais", true);
    const l = new Libro("POO en Java", "Autor X", 250);
    const pc = new Computador("Dell", 16, false);

    p.mostrarInfo();
    c.mostrarInfo();
    a.mostrarInfo();
    l.mostrarInfo();
    pc.mostrarInfo();
    console.log();
  }
}

File: Codigo de los 4 lenguajes/C#/EjemploClases.cs
using System;

public class Persona
{
    public string Nombre;
    public int Edad;
    public bool Activo;
    public Persona(string nombre,int edad,bool activo){ Nombre=nombre; Edad=edad; Activo=activo; }
    public void MostrarInfo(){ Console.WriteLine($"Persona -> nombre: {Nombre}, edad: {Edad}, activo: {Activo}"); }
}

public class Carro
{
    public string Marca;
    public int Modelo;
    public double Precio;
    public Carro(string marca,int modelo,double precio){ Marca=marca; Modelo=modelo; Precio=precio; }
    public void MostrarInfo(){ Console.WriteLine($"Carro  marca: {Marca}, modelo: {Modelo}, precio: {Precio}"); }
}

public class Animal
{
    public string Especie;
    public string Nombre;
    public bool Domestico;
    public Animal(string especie,string nombre,bool domestico){ Especie=especie; Nombre=nombre; Domestico=domestico; }
    public void MostrarInfo(){ Console.WriteLine($"Animal  especie: {Especie}, nombre: {Nombre}, domestico: {Domestico}"); }
}

public class Libro
{
    public string Titulo;
    public string Autor;
    public int Paginas;
    public Libro(string titulo,string autor,int paginas){ Titulo=titulo; Autor=autor; Paginas=paginas; }
    public void MostrarInfo(){ Console.WriteLine($"Libro -> titulo: {Titulo}, autor: {Autor}, paginas: {Paginas}"); }
}

public class Computador
{
    public string Marca;
    public int RamGb;
    public bool Encendido;
    public Computador(string marca,int ramGb,bool encendido){ Marca=marca; RamGb=ramGb; Encendido=encendido; }
    public void MostrarInfo(){ Console.WriteLine($"Computador -> marca: {Marca}, RAM: {RamGb}GB, encendido: {Encendido}"); }
}

public class EjemploClases
{
    public void Mostrar()
    {
        Console.WriteLine(" Ejemplo: Clases reales ");
        Persona p = new Persona("Camilo",20,true);
        Carro c = new Carro("Toyota",2020,35000.0);
        Animal a = new Animal("Perro","Firulais",true);
        Libro l = new Libro("POO en Java","Autor X",250);
        Computador pc = new Computador("Dell",16,false);

        p.MostrarInfo();
        c.MostrarInfo();
        a.MostrarInfo();
        l.MostrarInfo();
        pc.MostrarInfo();
        Console.WriteLine();
    }
}
